lms/static/js/dashboard/legacy.js

```diff
@@ -105,19 +105,48 @@
          });
 
          $('.action-unenroll').click(function(event) {
+
              var element = $(event.target);
-             var track_info = element.data('track-info');
+
+             var can_unenroll = $('#course-context-data').attr('data-course_can_unenroll');
+             var is_paid_course = $('#course-context-data').attr('data-course_is_paid_course');
+             var is_course_blocked = $('#course-context-data').attr('data-course_is_course_blocked');
+             var cert_name_long = $('#course-context-data').attr('data-course_cert_name_long');
+             var enrollment_mode = $('#course-context-data').attr('data-course_enrollment_mode');
+
+             var course_id = element.data('course_id');
              var course_number = element.data('course-number');
              var course_name = element.data('course-name');
-             var cert_name_long = element.data('cert-name-long');
-             $('#track-info').html(interpolate(track_info, {
+
+             var show_refund_option = "";
+
+             request = $.ajax({
+                      url: "/get_course_refund_status/?course_id"+course_id,
+                      method: "GET",
+                      dataType: "json"
+                    });
+
+                request.success(function( data ) {
+                    show_refund_option = data.course_refundable_status;
+                });
+                request.fail(function( jqXHR, textStatus ) {
+                    console.log(jqXHR);
+                });
+
+             var dialog_message_attr = set_dialog_attributes(can_unenroll, is_paid_course, is_course_blocked, cert_name_long,
```

Comment:
@adeelkhan Remember the request will generate a call back into javascript.
So, at this moment, the `show_refund_option` variable won't be defined. Only after the request comes back, it will be defined. You may need to wait until the api call request come back